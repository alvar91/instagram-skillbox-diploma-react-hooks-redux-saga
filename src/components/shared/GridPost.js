import React from 'react';
import PropTypes from 'prop-types';
import { useGridPostStyles } from '../../styles';
import { Typography } from '@material-ui/core';
import { useHistory } from 'react-router-dom';
import { getFormattedDate } from "../../utils/dateFormat";

function GridPost({ post }) {
  const { urls: {small: media} } = post;
  const history = useHistory();
  const classes = useGridPostStyles();

  function handleOpenPostModal() {
    history.push({
      pathname: `/${post.id}`,
      state: { modal: true },
    });
  }

  return (
    <div onClick={handleOpenPostModal} className={classes.gridPostContainer}>
      <div className={classes.gridPostOverlay}>
        <div className={classes.gridPostInfo}>
          <Typography>{getFormattedDate(post.updated_at)}</Typography>
        </div>
      </div>
      <img src={media} alt="Post cover" className={classes.image} />
    </div>
  );
}

GridPost.propTypes = {
  post: PropTypes.object.isRequired,
};


export default GridPost;
