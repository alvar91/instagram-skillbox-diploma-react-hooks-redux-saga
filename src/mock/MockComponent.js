import React from "react";

function MockComponent() {
  return <h2>Mock Component</h2>;
}

export default MockComponent;
