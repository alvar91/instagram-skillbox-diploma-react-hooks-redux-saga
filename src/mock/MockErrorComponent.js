import React from "react";

function MockErrorComponent() {
  React.useEffect(() => {
    throw new Error("Mock error");
  }, []);

  return <h2>Mock error</h2>;
}

export default MockErrorComponent;
