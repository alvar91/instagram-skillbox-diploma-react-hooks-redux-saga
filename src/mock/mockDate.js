export const mockDate = "2021-02-11T17:25:19-05:00";
export const correctMockDate = "Fri, Feb 12, 2021";
