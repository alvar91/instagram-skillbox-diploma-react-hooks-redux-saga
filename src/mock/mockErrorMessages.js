export const mockAuthErrorMessage =
  "Failed to get access token. Please try to log in again";
export const mockErrorMeta = "meta error";
