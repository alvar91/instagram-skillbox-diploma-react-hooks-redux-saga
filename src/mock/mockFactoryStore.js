/**
 * Function to create a mock store with input options
 * @param {object} storeOptionsObj contains options to create the mock store
 * @returns {object} mock store
 */
export const mockFactoryStore = ({postsObj = {}, usersObj = {}, authObj = {}}) => {
  const {
    posts = [],
    currentPage = 1,
    isFetchingPost = false,
    hasErrorPosts = false,
    errorMessagePosts = "",
  } = postsObj;

  const { users = [] } = usersObj;

  const {
    isFetchingAuth = false,
    isLoggedIn = false,
    profile = [],
    hasErrorAuth = false,
    errorMessageAuth = "",
  } = authObj;

  return {
    /* posts state */
    posts: {
      isFetchingPost: isFetchingPost,
      currentPage: currentPage,
      posts: posts,
      users: users,
      hasError: hasErrorPosts,
      errorMessage: errorMessagePosts,
    },
    /* auth state */
    auth: {
      isFetchingAuth: isFetchingAuth,
      isLoggedIn: isLoggedIn,
      profile: profile,
      hasError: hasErrorAuth,
      errorMessage: errorMessageAuth,
    },
  };
};
