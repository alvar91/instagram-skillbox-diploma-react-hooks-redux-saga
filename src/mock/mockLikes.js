export const mockId = 1;
export const mockCountLikes = 1;
