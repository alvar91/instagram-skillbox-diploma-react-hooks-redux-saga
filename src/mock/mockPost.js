export const mockPost = {
  id: "1gLdTsX3_70",
  created_at: "2020-07-01T18:30:14-04:00",
  updated_at: "2021-02-11T23:12:42-05:00",
  promoted_at: null,
  width: 6016,
  height: 4016,
  color: "#260c0c",
  blur_hash: "LHEf7wbbM{w[_NM{-:xZRiVs%gNG",
  description: null,
  alt_description: "laptop on brown sofa",
  urls: {
    raw:
      "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1",
    full:
      "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?crop=entropy&cs=srgb&fm=jpg&ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=85",
    regular:
      "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=80&w=1080",
    small:
      "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=80&w=400",
    thumb:
      "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=80&w=200",
  },
  links: {
    self: "https://api.unsplash.com/photos/1gLdTsX3_70",
    html: "https://unsplash.com/photos/1gLdTsX3_70",
    download: "https://unsplash.com/photos/1gLdTsX3_70/download",
    download_location: "https://api.unsplash.com/photos/1gLdTsX3_70/download",
  },
  categories: [],
  likes: 738,
  liked_by_user: false,
  current_user_collections: [],
  sponsorship: {
    impression_urls: [
      "https://secure.insightexpressai.com/adServer/adServerESI.aspx?script=false&bannerID=8281535&rnd=[timestamp]&gdpr=&gdpr_consent=&redir=https://secure.insightexpressai.com/adserver/1pixel.gif",
    ],
    tagline: "Designed to be the Best",
    tagline_url: "http://www.dell.com/xps",
    sponsor: {
      id: "2DC3GyeqWjI",
      updated_at: "2021-02-12T14:33:51-05:00",
      username: "xps",
      name: "XPS",
      first_name: "XPS",
      last_name: null,
      twitter_username: "Dell",
      portfolio_url: "http://www.dell.com/xps",
      bio:
        "Designed to be the best, with cutting edge technologies, exceptional build quality, unique materials and powerful features.",
      location: null,
      links: {
        self: "https://api.unsplash.com/users/xps",
        html: "https://unsplash.com/@xps",
        photos: "https://api.unsplash.com/users/xps/photos",
        likes: "https://api.unsplash.com/users/xps/likes",
        portfolio: "https://api.unsplash.com/users/xps/portfolio",
        following: "https://api.unsplash.com/users/xps/following",
        followers: "https://api.unsplash.com/users/xps/followers",
      },
      profile_image: {
        small:
          "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32",
        medium:
          "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64",
        large:
          "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128",
      },
      instagram_username: "dell",
      total_collections: 0,
      total_likes: 0,
      total_photos: 22,
      accepted_tos: true,
    },
  },
  user: {
    id: "2DC3GyeqWjI",
    updated_at: "2021-02-12T14:33:51-05:00",
    username: "xps",
    name: "XPS",
    first_name: "XPS",
    last_name: null,
    twitter_username: "Dell",
    portfolio_url: "http://www.dell.com/xps",
    bio:
      "Designed to be the best, with cutting edge technologies, exceptional build quality, unique materials and powerful features.",
    location: null,
    links: {
      self: "https://api.unsplash.com/users/xps",
      html: "https://unsplash.com/@xps",
      photos: "https://api.unsplash.com/users/xps/photos",
      likes: "https://api.unsplash.com/users/xps/likes",
      portfolio: "https://api.unsplash.com/users/xps/portfolio",
      following: "https://api.unsplash.com/users/xps/following",
      followers: "https://api.unsplash.com/users/xps/followers",
    },
    profile_image: {
      small:
        "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32",
      medium:
        "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64",
      large:
        "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128",
    },
    instagram_username: "dell",
    total_collections: 0,
    total_likes: 0,
    total_photos: 22,
    accepted_tos: true,
  },
};
