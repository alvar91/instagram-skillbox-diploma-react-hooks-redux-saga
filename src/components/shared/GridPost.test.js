// Core
import React from "react";
import { configure, mount } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import GridPost from "./GridPost";
import { mockPost } from "../../mock/mockPost";

configure({ adapter: new Adapter() });

/**
 * Setup function for GridPost component
 * @param {object} storeOptionsObj contains options for creating the mock store
 * @returns {ReactRenderer}
 */
const setupTestRenderer = () => {
  return mount(<GridPost post={ mockPost } />);
};

describe("GridPost", () => {
  describe("GridPost with mock post state", () => {
    let testRenderer;
    let component;
    const props = { post: mockPost };

    beforeAll(() => {
      testRenderer = setupTestRenderer();
      component = testRenderer.find(GridPost);
    });

    it("GridPost should correctly renders markup", () => {
      expect(shallowToJson(testRenderer)).toMatchSnapshot();
    });

    it("GridPost should contain correct props", () => {
      expect(component).toHaveLength(1);
      expect(component.props()).toMatchObject(props);
    });
  });
});
