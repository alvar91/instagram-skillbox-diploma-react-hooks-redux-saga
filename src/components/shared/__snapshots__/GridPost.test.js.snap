// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridPost GridPost with mock post state GridPost should correctly renders markup 1`] = `
<GridPost
  post={
    Object {
      "alt_description": "laptop on brown sofa",
      "blur_hash": "LHEf7wbbM{w[_NM{-:xZRiVs%gNG",
      "categories": Array [],
      "color": "#260c0c",
      "created_at": "2020-07-01T18:30:14-04:00",
      "current_user_collections": Array [],
      "description": null,
      "height": 4016,
      "id": "1gLdTsX3_70",
      "liked_by_user": false,
      "likes": 738,
      "links": Object {
        "download": "https://unsplash.com/photos/1gLdTsX3_70/download",
        "download_location": "https://api.unsplash.com/photos/1gLdTsX3_70/download",
        "html": "https://unsplash.com/photos/1gLdTsX3_70",
        "self": "https://api.unsplash.com/photos/1gLdTsX3_70",
      },
      "promoted_at": null,
      "sponsorship": Object {
        "impression_urls": Array [
          "https://secure.insightexpressai.com/adServer/adServerESI.aspx?script=false&bannerID=8281535&rnd=[timestamp]&gdpr=&gdpr_consent=&redir=https://secure.insightexpressai.com/adserver/1pixel.gif",
        ],
        "sponsor": Object {
          "accepted_tos": true,
          "bio": "Designed to be the best, with cutting edge technologies, exceptional build quality, unique materials and powerful features.",
          "first_name": "XPS",
          "id": "2DC3GyeqWjI",
          "instagram_username": "dell",
          "last_name": null,
          "links": Object {
            "followers": "https://api.unsplash.com/users/xps/followers",
            "following": "https://api.unsplash.com/users/xps/following",
            "html": "https://unsplash.com/@xps",
            "likes": "https://api.unsplash.com/users/xps/likes",
            "photos": "https://api.unsplash.com/users/xps/photos",
            "portfolio": "https://api.unsplash.com/users/xps/portfolio",
            "self": "https://api.unsplash.com/users/xps",
          },
          "location": null,
          "name": "XPS",
          "portfolio_url": "http://www.dell.com/xps",
          "profile_image": Object {
            "large": "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128",
            "medium": "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64",
            "small": "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32",
          },
          "total_collections": 0,
          "total_likes": 0,
          "total_photos": 22,
          "twitter_username": "Dell",
          "updated_at": "2021-02-12T14:33:51-05:00",
          "username": "xps",
        },
        "tagline": "Designed to be the Best",
        "tagline_url": "http://www.dell.com/xps",
      },
      "updated_at": "2021-02-11T23:12:42-05:00",
      "urls": Object {
        "full": "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?crop=entropy&cs=srgb&fm=jpg&ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=85",
        "raw": "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1",
        "regular": "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=80&w=1080",
        "small": "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=80&w=400",
        "thumb": "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=80&w=200",
      },
      "user": Object {
        "accepted_tos": true,
        "bio": "Designed to be the best, with cutting edge technologies, exceptional build quality, unique materials and powerful features.",
        "first_name": "XPS",
        "id": "2DC3GyeqWjI",
        "instagram_username": "dell",
        "last_name": null,
        "links": Object {
          "followers": "https://api.unsplash.com/users/xps/followers",
          "following": "https://api.unsplash.com/users/xps/following",
          "html": "https://unsplash.com/@xps",
          "likes": "https://api.unsplash.com/users/xps/likes",
          "photos": "https://api.unsplash.com/users/xps/photos",
          "portfolio": "https://api.unsplash.com/users/xps/portfolio",
          "self": "https://api.unsplash.com/users/xps",
        },
        "location": null,
        "name": "XPS",
        "portfolio_url": "http://www.dell.com/xps",
        "profile_image": Object {
          "large": "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128",
          "medium": "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64",
          "small": "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32",
        },
        "total_collections": 0,
        "total_likes": 0,
        "total_photos": 22,
        "twitter_username": "Dell",
        "updated_at": "2021-02-12T14:33:51-05:00",
        "username": "xps",
      },
      "width": 6016,
    }
  }
>
  <div
    className="makeStyles-gridPostContainer-2"
    onClick={[Function]}
  >
    <div
      className="makeStyles-gridPostOverlay-3"
    >
      <div
        className="makeStyles-gridPostInfo-4"
      >
        <WithStyles(ForwardRef(Typography))>
          <ForwardRef(Typography)
            classes={
              Object {
                "alignCenter": "MuiTypography-alignCenter",
                "alignJustify": "MuiTypography-alignJustify",
                "alignLeft": "MuiTypography-alignLeft",
                "alignRight": "MuiTypography-alignRight",
                "body1": "MuiTypography-body1",
                "body2": "MuiTypography-body2",
                "button": "MuiTypography-button",
                "caption": "MuiTypography-caption",
                "colorError": "MuiTypography-colorError",
                "colorInherit": "MuiTypography-colorInherit",
                "colorPrimary": "MuiTypography-colorPrimary",
                "colorSecondary": "MuiTypography-colorSecondary",
                "colorTextPrimary": "MuiTypography-colorTextPrimary",
                "colorTextSecondary": "MuiTypography-colorTextSecondary",
                "displayBlock": "MuiTypography-displayBlock",
                "displayInline": "MuiTypography-displayInline",
                "gutterBottom": "MuiTypography-gutterBottom",
                "h1": "MuiTypography-h1",
                "h2": "MuiTypography-h2",
                "h3": "MuiTypography-h3",
                "h4": "MuiTypography-h4",
                "h5": "MuiTypography-h5",
                "h6": "MuiTypography-h6",
                "noWrap": "MuiTypography-noWrap",
                "overline": "MuiTypography-overline",
                "paragraph": "MuiTypography-paragraph",
                "root": "MuiTypography-root",
                "srOnly": "MuiTypography-srOnly",
                "subtitle1": "MuiTypography-subtitle1",
                "subtitle2": "MuiTypography-subtitle2",
              }
            }
          >
            <p
              className="MuiTypography-root MuiTypography-body1"
            >
              Fri, Feb 12, 2021
            </p>
          </ForwardRef(Typography)>
        </WithStyles(ForwardRef(Typography))>
      </div>
    </div>
    <img
      alt="Post cover"
      className="makeStyles-image-1"
      src="https://images.unsplash.com/photo-1593642532454-e138e28a63f4?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=80&w=400"
    />
  </div>
</GridPost>
`;
